safeeyes/utility.py

```diff
@@ -94,28 +94,16 @@ def start_thread(target_function, **args):
     thread.start()
 
 
-# def execute_main_thread(target_function, args=None):
+# def execute_main_thread(target_function, *args, **kwargs):
 #     """
-#     Execute the given function in main thread.
+#     Execute the given function in main thread, forwarding positional and keyword arguments.
 #     """
-#     if args:
-#         GLib.idle_add(lambda: target_function(args))
-#     else:
-#         GLib.idle_add(target_function)
 
-def execute_main_thread(target_function, arg1=None, arg2=None):
+def execute_main_thread(target_function, *args, **kwargs):
     """
     Execute the given function in main thread.
     """
-    if arg1 is not None and arg2 is not None:
-        GLib.idle_add(lambda: target_function(arg1, arg2))
-    elif arg1 is not None:
-        GLib.idle_add(lambda: target_function(arg1))
-    elif arg2 is not None:
-        GLib.idle_add(lambda: target_function(arg2))
-    else:
-        GLib.idle_add(target_function)
-
+    GLib.idle_add(lambda: target_function(*args, **kwargs))
```

Comment:
This change is technically not necessary to fix the crash - however, the smartpause plugin calls `disable_safeeyes(None, True)`. With the previous code, this would see that `arg1` is `None` - and pass `arg2` to the first argument, instead.
This lead to SafeEyes being "manually" disabled, meaning it would not be re-enabled when the system was no longer idle, only when the user clicked "Enable Safe Eyes".

In theory, this could break any caller of `execute_main_thread` that relied on the older, less intuitive behaviour.
However, `disable_safeeyes` is the only call that uses the second argument. `enable_safeeyes` has a second argument, but that is not used anywhere.